src/pages/loginOrRegister.js

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import "./loginOrRegister.css";
+import { LoginForm } from "../components/loginForm";
+import { RegisterForm } from "../components/registerForm";
+
+function LoginOrRegister(props) {
+  const [activeForm, setActiveForm] = useState("login");
+  const [message, setMessage] = useState("");
+
+  let showLoginForm = (m = "") => {
+    setActiveForm("login");
+    if (typeof m === "string") setMessage(m);
+  };
+
+  let showRegisterForm = () => {
+    setActiveForm("register");
+    setMessage("");
+  };
+  const loginForm = <LoginForm setLoggedIn={props.setLoggedIn} />;
+  const registerForm = <RegisterForm changePage={showLoginForm} />;
```

Comment:
You are initializing both `LoginForm` and `RegisterForm` components and at the end, one will be discarded. You can initialize one component after taking a decision.